Automattic/WP-Job-Manager
# Fix Salary Field Currency and Unit

Fixes #2278

### Changes proposed in this Pull Request

* Add an option to set default salary currency on the settings panel;
* Add an option to set the default salary unit (or period of time, I may need to find a better name here) on the settings panel;
* Add an option to enable or disable customization of the salary currency on the settings panel for each job;
* Add an option to enable or disable customization of the salary unit (again, of period..) on the settings panel for each job;
* Add, on the job editor, an option to set salary currency (if that's enabled on the settings);
* Add, on the job editor, an option to set salary unit (if that's enabled on the settings);
* Allow setting an empty salary unit, which disables the addition of a "/" on the salary. Might need to document this behavior better;
* If there's no salary, hide the icon in the frontend;
* Add the new fields to the frontend, so users submitting new jobs or editing already registered jobs can see it too;
* Fix an issue with non-translated job units in the frontend; 

### Testing instructions

1. Install the plugin containing this branch of the code in a WordPress install;
2. Check the new options in Job Listings -> Settings -> Job Listings related to Salary Currency and Unit;
3. Verify the new fields related to Salary Currency and Unit on the page Job Listings -> Add New and their behavior;
4. Verify the new information appearing on the Job Page;

### New Hooks

- `the_job_salary_message` - Filter the returned job salary message (with the additional formatting);
- `wpjm_job_salary_unit_display_text` - Filter the returned job salary unit display text (for translations and other messages);

### Tasks to do

- [x] Right now, if you enable the customization of the salary currency, and edit the salary currency for a given job, and then disable the customization again, the job will keep showing the customized salary currency. I'm thinking here about possible issues with this approach;
- [x] Verify the output of the structured data in each case;
- [x] Check if the new hook is indeed necessary;
